src/router/ProtectedRouter.tsx

```diff
@@ -1,11 +1,24 @@
-import { Outlet } from "react-router-dom";
+import { Navigate, Outlet } from "react-router-dom";
 import LandingPage from "../pages/LandingPage/LandingPage";
 import { useAuthContext } from "../store/auth/AuthContext";
+import { userRoles } from "../constants/userRoles";
+import { routes } from "../constants/routes";
 
-const ProtectedRouter = () => {
-  const { isLoggedIn } = useAuthContext();
+type Props = {
+  role?: string;
+};
+
+const ProtectedRouter = ({ role = userRoles.ROLE_USER }: Props) => {
+  const { isLoggedIn, isUserAdmin } = useAuthContext();
+
+  if (isLoggedIn) {
```

Comment:
Maybe we can invert logic here and check if user is not logged in
That way we wouldn't have nested if's